### PR TITLE
Remove EOL PHP 5.4 from `.travis.yml` and `composer.json` - Fixes #5862

### LINKED ISSUE #5862: Remove PHP 5.4 from travis builds
PHP 5.4 has reached end of life I don't understand why the travis run for PHP 5.4 is still required. See https://travis-ci.org/doctrine/doctrine2/jobs/135986241

It currently breaks travis builds for new PRs, i.e. the UnitOfWorkTest is using the `Country::class` notation which was introduced in PHP 5.5 (which will actually reach EOL very soon, too).

Not sure if making `UnitOfWorkTest` use `get_class()` for now, or just remove PHP 5.4 from the travis build.

```
$ php -v
PHP 5.4.40-1~dotdeb+wheezy.1 (cli) (built: Apr 22 2015 06:50:56) 
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2014 Zend Technologies
    with Xdebug v2.3.2, Copyright (c) 2002-2015, by Derick Rethans
    with Zend OPcache v7.0.3-dev, Copyright (c) 1999-2013, by Zend Technologies
$ ./vendor/bin/phpunit tests/Doctrine/Tests/ORM/UnitOfWorkTest.php 
PHP Parse error:  syntax error, unexpected 'class' (T_CLASS), expecting identifier (T_STRING) or variable (T_VARIABLE) or '{' or '$' in /vagrant/libs/doctrine2/tests/Doctrine/Tests/ORM/UnitOfWorkTest.php on line 343
PHP Stack trace:
PHP   1. {main}() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/phpunit:0
PHP   2. PHPUnit_TextUI_Command::main() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/phpunit:47
PHP   3. PHPUnit_TextUI_Command->run() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/src/TextUI/Command.php:100
PHP   4. PHPUnit_Runner_BaseTestRunner->getTest() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/src/TextUI/Command.php:123
PHP   5. PHPUnit_Runner_BaseTestRunner->loadSuiteClass() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/src/Runner/BaseTestRunner.php:67
PHP   6. PHPUnit_Runner_StandardTestSuiteLoader->load() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/src/Runner/BaseTestRunner.php:123
PHP   7. PHPUnit_Util_Fileloader::checkAndLoad() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/src/Runner/StandardTestSuiteLoader.php:39
PHP   8. PHPUnit_Util_Fileloader::load() /vagrant/libs/doctrine2/vendor/phpunit/phpunit/src/Util/Fileloader.php:38
```

### PR DESCRIPTION
**Pull request by tPl0ch**
See https://github.com/doctrine/doctrine2/issues/5862 for more info

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     ],
     "minimum-stability": "dev",
     "require": {
-        "php": ">=5.4",
+        "php": ">=5.5",
```
Comment (Ocramius) [inline composer.json]: We should change this to `^5.5 || ^7.0`

**Comment by tPl0ch** [comment]
@Ocramius Done

**Comment by Soullivaneuh** [comment]
@Ocramius :+1: for version support dropping, but this should be done on a new major IMHO.

Dropping PHP support is BC break for developer using this bundle under the dropped version.

**Comment by Ocramius** [comment]
> Dropping PHP support is BC break

This has been discussed ad-nauseam. Dependency requirement bumps (*any dependency*) are not BC breaks. This will not break running systems, and systems running 5.4 will simply reject Doctrine ORM 2.6.x when running `composer update`, falling back to 2.5.x.

We've also done this previously in the 2.4 -> 2.5 upgrade.

 >  should be done on a new major IMHO.

Done it multiple times in Doctrine, in Zendframework and in other libraries: not a problem, and also not perceived as a problem by consumers. Dropping support for un-maintained PHP versions is also the most responsible thing to do from a security perspective.

**Comment by Ocramius** [comment]
Failing tests are related to #5854 and #5856

**Comment by Ocramius** [comment]
Thanks, @tPl0ch!

**Comment by Soullivaneuh** [comment]
@Ocramius I understand, you are technically right.

Symfony does not follows this way, so I think it's a matter of choices...

Thanks for your explanation anyway. :+1:

**Comment by Ocramius** [comment]
> Symfony does not follows this way, so I think it's a matter of choices...

Top of the page says `doctrine/`, in fact, not `symfony/` :-P
Also, we don't have the throughput capabilities of Symfony: less versions to maintain is just less headaches.

**Comment by Soullivaneuh** [comment]
> Top of the page says `doctrine/`, in fact, not `symfony/` :-P

Yeas I know, this is was I said:

> so I think it's a matter of choices

:wink: 

> Also, we don't have the throughput capabilities of Symfony: less versions to maintain is just less headaches.

I totally understand you for this point. :+1:  Having same issue for Sonata projects.

**Comment by tPl0ch** [comment]
@Soullivaneuh In addition to @Ocramius points I also believe that libraries can be a vital driver for new PHP version adoptions this way. When an EOL is reached it's time to increase the pressure on consumers, and libraries like doctrine can instrument a big leverage here.

**Pull request merged**